### PR TITLE
[ROX-13346] Add autocomplete option for new vulnerability search categories in graphQL

### PR DESCRIPTION
**Pull request by md2119**
## Description

Apparently, graphQL and gRPC endpoints maintain separate implementations for auto-complete. This PR updates the graphQL autocomplete that was missed.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

File: central/graphql/resolvers/search.go [hunk 1]
```diff
@@ -139,6 +139,11 @@ func (resolver *Resolver) getAutoCompleteSearchers() map[v1.SearchCategory]searc
 	}
 	if !features.PostgresDatastore.Enabled() {
 		searchers[v1.SearchCategory_VULNERABILITIES] = resolver.CVEDataStore
+	} else {
+		searchers[v1.SearchCategory_IMAGE_VULNERABILITIES] = resolver.ImageCVEDataStore
+		searchers[v1.SearchCategory_NODE_VULNERABILITIES] = resolver.NodeCVEDataStore
+		searchers[v1.SearchCategory_CLUSTER_VULNERABILITIES] = resolver.ClusterCVEDataStore
+		searchers[v1.SearchCategory_NODE_COMPONENTS] = resolver.NodeComponentDataStore
```
Comment (vjwilson) [inline central/graphql/resolvers/search.go:146]: Do we need to add a search for `SearchCategory_IMAGE_COMPONENTS` ?

**Review by vjwilson — COMMENTED** [review]
I will leave others to comment on the implementation; I just had one question about the searchers implemented.

File: central/graphql/resolvers/search.go (hunk 1 above)
Comment (md2119) [reply central/graphql/resolvers/search.go:146]: `ln137`

**Review by vjwilson — APPROVED** [review]

File: central/globalindex/mapping/mapping.go [hunk 2]
```diff
@@ -146,24 +160,27 @@ func GetEntityOptionsMap() map[v1.SearchCategory]search.OptionsMap {
...
 		v1.SearchCategory_ROLEBINDINGS:          roleBindingOptions.OptionsMap,
 		v1.SearchCategory_SERVICE_ACCOUNTS:      serviceAccountOptions.OptionsMap,
 		v1.SearchCategory_SUBJECTS:              subjectMapping.OptionsMap,
-		v1.SearchCategory_VULNERABILITIES:       cveSearchOptions,
-		v1.SearchCategory_COMPONENT_VULN_EDGE:   componentVulnEdgeMapping.OptionsMap,
-		v1.SearchCategory_CLUSTER_VULN_EDGE:     clusterVulnEdgeMapping.OptionsMap,
-		v1.SearchCategory_IMAGE_COMPONENT_EDGE:  imageComponentEdgeMapping.OptionsMap,
-		v1.SearchCategory_IMAGE_COMPONENTS:      componentSearchOptions,
-		v1.SearchCategory_IMAGE_VULN_EDGE:       imageCVEEdgeMapping.OptionsMap,
-		v1.SearchCategory_NODE_COMPONENT_EDGE:   nodeComponentEdgeMappings.OptionsMap,
```
Comment (md2119) [inline central/globalindex/mapping/mapping.go:163]: Edges should not be exposed.

Comment (connorgorman) [reply central/globalindex/mapping/mapping.go:163]: This is within GetEntityOptionsMap which gets called from here:
```
		for category, optMap := range GetEntityOptionsMap() {
			indexMapping.AddDocumentMapping(category.String(), blevesearch.DocumentMappingFromOptionsMap(optMap.Original()))
		}
```
Does this impact the persisted index in dackbox?

Comment (md2119) [reply central/globalindex/mapping/mapping.go]: not relevant anymore

**Comment by md2119** [comment]
Manually tested autocomplete works on image, node, and cluster cves page:
image
image
image

**Pull request merged**